subprojects/core/src/main/java/org/gradle/api/internal/initialization/ResettableConfiguration.java

```diff
@@ -47,6 +52,7 @@ public interface ResettableConfiguration extends Configuration {
      * @implSpec Usage: This method should only be called on resolvable configurations and should throw an exception if
      * called on a configuration that does not permit this usage.
      */
-    void resetResolutionState();
+    @Nullable
+    <T> T withResetResolutionState(Factory<T> factory);
```

Comment:
Rename to `callAndResetResolutionState`.

Reply:
Done